Rules/CommonScripts/MapVotesCommon.as

```diff
@@ -0,0 +1,610 @@
+					done = true;
+				}
+				else if (temp != button3.filename)
+				{
+					button1.filename = temp;
+					button1.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(temp));
+					done = true;
+				}
+				else if (count == arrayleng) //safely exit the loop, pure random, probably only 1 map in the pool
+				{
+					button1.filename = map_names[_random.NextRanged(arrayleng)];
+					button1.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(button1.filename));
+					done = true;
+				}
+			}
+
+			//randomize button 3 map name
+			count = 0;
+			done = false; 
+			while (!done)
+			{
+				count++;
+				string temp = map_names[_random.NextRanged(arrayleng)];
+
+				string temptest = temp.substr(temp.length() - 1,temp.length() - 1);
+				if (temptest == ")")
+				{
+					string[] name = temp.split(' (');
+					string mapName = name[name.length() - 1];
+					temp = mapName.substr(0,mapName.length() - 1);
+				}
+
+				if (arrayleng == 2)
+				{
+					button3.filename = map_names[1];
+					button3.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(button3.filename));
+					done = true;
+				}
+				else if (temp != button3.filename && temp != button1.filename && temp != currentMap)
+				{
+					button3.filename = temp;
+					button3.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(temp));
+					done = true;
+				}
+				else if (temp != button3.filename && temp != button1.filename)
+				{
+					button3.filename = temp;
+					button3.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(temp));
+					done = true;
+				}
+				else if (temp != button1.filename)
+				{
+					button3.filename = temp;
+					button3.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(temp));
+					done = true;
+				}
+				else if (count == arrayleng)
+				{
+					button3.filename = map_names[_random.NextRanged(arrayleng)];
+					button3.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(button3.filename));
+					done = true;
+				}
+			}
+		}
+	}
+
+	void RefreshButtons()
+	{	
+		Vec2f ButtonSize;
+		MenuSize.x = 30;
+		MenuSize.y = 200;
+
+		button1.RefreshButton( MenuSize.x, ButtonSize);
+		MenuSize.x += ButtonSize.x+30;
+		MenuSize.y = (ButtonSize.y+100) > MenuSize.y ? ButtonSize.y+95 : MenuSize.y;
+
+		button2.RefreshRandomButton( MenuSize.x, ButtonSize);
+		MenuSize.x += ButtonSize.x+30;
+		MenuSize.y = (ButtonSize.y+100) > MenuSize.y ? ButtonSize.y+95 : MenuSize.y;
+		
+		button3.RefreshButton( MenuSize.x, ButtonSize);
+		MenuSize.x += ButtonSize.x+30;
+		MenuSize.y = (ButtonSize.y+100) > MenuSize.y ? ButtonSize.y+95 : MenuSize.y;
+	}
+
+	void Update(CControls@ controls, u8 &out SelectedNum)
+	{
+		Vec2f mousepos = controls.getMouseScreenPos();
+		const bool mousePressed = controls.isKeyPressed(KEY_LBUTTON);
+		const bool mouseJustReleased = controls.isKeyJustReleased(KEY_LBUTTON);
+
+		int hoverednum;
+		if (button1.isHovered(mousepos))
+		{
+			if (button1.State == 3) return; 
```

Comment:
perhaps we should use enums for consts for the button states so this code is more readable.